packages/cozy-harvest-lib/src/models/ConnectionFlow.js

```diff
@@ -483,9 +486,16 @@ export class ConnectionFlow {
     }
   }
 
-  async handleJobUpdated() {
+  handleCurrentJobUpdated() {
     logger.debug('ConnectionFlow: Handling update from job')
-    await this.refetchTrigger()
+    this.refetchTrigger()
+  }
+
+  handleTriggerJobUpdated(job) {
```

Comment:
nit: I'll have created two methods: 

- listenJobsEvent 
- And this listenJobsEvent will have passed down the `job` to `handleJobsUpdateForCurrentTrigger` or something similar.